src/prefect/utilities/logging.py

```diff
@@ -30,6 +33,57 @@ def __init__(self) -> None:
         self.logger.addHandler(handler)
         self.logger.setLevel(context.config.logging.level)
 
+    @property
+    def queue(self) -> Queue:
+        if not hasattr(self, "_queue"):
+            self._queue = Queue()  # type: Queue
+            self._flush = False
+            self.start()
+        return self._queue
+
+    def flush(self) -> None:
+        self._flush = True
+        if self.client is not None:
+            self.batch_upload()
+            self._thread.join()
+
+    def batch_upload(self) -> None:
+        logs = []
+        try:
+            while True:
+                log = self.queue.get(False)
+                logs.append(log)
```

Comment:
There's some (small) risk that the arrival rate of logs in `self.queue` is > the processing rate of this function, in which case it might never leave this `while True` loop. Perhaps a `max_iter` counter just in case?